bare_metal_tests/TESTS/bringup/uart/test.py

```diff
@@ -0,0 +1,330 @@
+    file.close()
+
+def run_command(cmd):
+    """! Runs command and prints proc stdout on screen.
+         It is used to build the test.
+         
+    @param cmd command to be executed.
+    @return image path.
+    """
+    image_path = str()
+    
+    # Remove build log file if exists
+    try:
+        os.remove('build.log')
+    except OSError:
+        pass
+    
+    try:
+        p = Popen(cmd, stdout=sys.stdout)
+    except OSError as e:
+        logging.error('Command can not be executed.')
+        exit()
+
+    # Check process status.
+    returncode = p.wait()
+    if returncode < 0:
+        logging.error('Error while executing command.')
+        exit()
+    
+    # Get image path     
+    try:
+        with open('build.log') as f:
+            print('file ok')
+            build_data = json.load(f)
+            print('json ok')
+            image_path = build_data['builds'][0]['bin']
+            print('json ok')
+    except:
+        logging.error('Unable to locate image file.')
+        exit()
+       
+    return image_path
+    
+def reset_mbed_board(port, delay_reset):
+    """Perform reset of the mbed device by sending
+       a break signal to the device.
+       
+    @param port serial port id.
+    @param delay_reset delay time after reset [s].
+    """
+    logging.info('Reset the mbed board...')
+ 
+    break_port = serial.Serial(port=port, baudrate=9600)
+    
+    break_port.sendBreak()
+
+    break_port.close()
+    
+    logging.info('Wait %f sec after the reset.' % (delay_reset))
+    time.sleep(delay_reset)
+    
+def flash_mbed_board(src, dst, delay):
+    """Perform flash of the mbed device.
+    
+    @param src path to the .bin file.
+    @param dst path to mbed device.
+    @param delay delay time after flashing [s].
+    """
+    logging.info('Flashing mbed board...')
+    logging.info('source: %s' % (src))
+    logging.info('destination: %s' % (dst))
+    
+    shutil.copyfile(src, dst)
+    
+    logging.info('Wait %f sec after flashing.' % (delay))
+    time.sleep(delay)
+    
+def build_test(target, toolchain, clean_flag):
+    """Build the mbed test.
+    
+    @param target target id.
+    @param toolchain compile toolchain.
+    @return image path.
+    """
+    logging.info('Building mbed UART test for green tea.')
+    
+    clean = ''
+    if(clean_flag):
+        clean = '-c'
+    
+    cmd = 'mbed test -t %s -m %s --source . --source bare_metal_tests --build-data bare_metal_tests/TESTS/bringup/uart/build.log --compile %s' % (toolchain, target, clean)
+    return run_command(cmd)
+
+def open_port(port_id, baudrate):
+    """Open serial port.
+    
+    @param port_id serial port id.
+    @param baudrate serial port baudrate.
+    """
+    global serial_port
+    try:
+        serial_port = serial.Serial(
+        port=port_id,
+        baudrate=baudrate,
+        parity=serial.PARITY_NONE,
+        stopbits=serial.STOPBITS_ONE,
+        bytesize=serial.EIGHTBITS)
```

Comment:
The indention of this line, and the prior 4 lines is incorrect. Could you fix it?

Reply:
Fixed.